ipywidgets/widgets/widget.py

```diff
@@ -638,15 +596,17 @@ def _should_send_property(self, key, value):
         """Check the property lock (property_lock)"""
         to_json = self.trait_metadata(key, 'to_json', self._trait_to_json)
         if key in self._property_lock:
-            # model_state, buffer_paths, buffers
-            split_value = _remove_buffers({ key: to_json(value, self)})
-            split_lock = _remove_buffers({ key: self._property_lock[key]})
+            buffers = []
+            if len(inspect.signature(to_json).parameters) > 2:
+                serializer_value = { key: to_json(value, self, buffers)}
+            else:
+                serializer_value = { key: to_json(value, self)}
+            split_lock = { key: self._property_lock[key]}
             # A roundtrip conversion through json in the comparison takes care of
             # idiosyncracies of how python data structures map to json, for example
             # tuples get converted to lists.
-            if (jsonloads(jsondumps(split_value[0])) == split_lock[0]
-                and split_value[1] == split_lock[1]
-                and _buffer_list_equal(split_value[2], split_lock[2])):
+            if (jsonloads(jsondumps(serializer_value)) == split_lock
+                and _buffer_list_equal(buffers, self._buffer_lock)):
```

Comment:
@SylvainCorlay This solution here is broken, since the buffers are not checked *per key*. There needs to be a way to check the buffers "owned" by a specific key/trait. E.g. by having a mapping (key -> buffer indices).